### PR TITLE
contrib/dimfeld/httptreemux.v5: parameterize redirects due to trailing slash

### PR DESCRIPTION
**Pull request by laughingman-hass**
### What does this PR do?

Adds support for routes that contain a trailing slash which result in a redirect. If a redirect is detected due to a trailing slash, look up the parameters for the route without the trailing slash and use that for the parameterization.

### Motivation

Bug raised https://github.com/DataDog/dd-trace-go/issues/2293

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Comment by laughingman-hass** [comment]
Hi DD folks 👋 

Just wanted to check if there is anything else needed from me to move this PR forward?

File: contrib/dimfeld/httptreemux.v5/httptreemux_test.go [hunk 1]
```diff
@@ -246,6 +303,7 @@ func router() http.Handler {
 
 	router.GET("/200", handler200)
 	router.GET("/500", handler500)
+	router.GET("/api/:parameter", handlerDummy)
```
[the remark this answers is not in the document]
Comment (laughingman-hass) [reply contrib/dimfeld/httptreemux.v5/httptreemux_test.go:305]: So this was a choice for a ease of readability and not to confuse it with a 200 response. In the trailing slash test the handler is ignored, hence calling it 'dummy'. The router detects that the request should be redirected and thats what is being tested. If there was a second request to follow the redirect then, the handler becomes more important.

**Comment by katiehockman** [comment]
@laughingman-hass thank you! I think the only thing needed is a quick rebase. For some reason I can't do it on my end. Mind giving that a quick try?

**Comment by laughingman-hass** [comment]
rebased `main` branch as requested

**Pull request merged**